pkg/kubeapiserver/options/storage_versions.go

```diff
@@ -103,6 +103,9 @@ func (s *StorageSerializationOptions) AddFlags(fs *pflag.FlagSet) {
 		"You only need to pass the groups you wish to change from the defaults. "+
 		"It defaults to a list of preferred versions of all known groups.")
 
+	fs.MarkDeprecated("storage-versions", "Please avoid using this flag. "+
```

Comment:
But if they have been using this flag, they need to run some form of storage migration to fix it before upgrading...

Reply:
> But if they have been using this flag, they need to run some form of storage migration to fix it before upgrading...

not really... whatever versions they were storing before *must* be able to be encoded/decoded by the apiserver, and we haven't removed the ability of the apiserver to decode any types it could decode in 1.11

Reply:
the point of removing this is so that we can know what is being encoded, so that in a future release we can tell them to run a storage migration tool, know versions x,y,z are stored, then safely remove versions a,b,c